### DISCUSSION TITLE
Adding a GUI

### DISCUSSION THREAD

**Discussion by shayded-exe** [rshea0/enjinn#32] [Ideas]
I'm considering adding a GUI after all. Something simple.

What do y'all think? What features would be useful for you?

**Comment by Houseman-HH** [comment]
Very good Idea- using command line is not my thing

**Comment by JONESYdj** [comment]
Great idea.
Basically something like this:
![](https://www.wearecrossfader.co.uk/wp-content/uploads/2019/04/Screenshot-2019-04-15-at-14.46.29-1024x538.png)

**Reply by shayded-exe** [reply]
The smart playlist rules will still be configured via text file like the [example here](https://github.com/rshea0/enjinn/blob/master/examples/enjinn.example.yaml).

Because of the ability to nest and group rules, it's not quite as simple as your example.

I will at a minimum have a button to open the config file in your text editor. Maybe I could allow you to edit the file directly within ENJINN. It wouldn't be too hard to bundle in a simple editor with validation. It still be a lot easier and faster than building out a whole rule editing UI.
